test/object-store/c_api/c_api.cpp

```diff
@@ -3962,6 +3962,163 @@ struct Userdata {
 };
 
 #if REALM_ENABLE_SYNC
+
+TEST_CASE("C API - client reset", "[c_api][client-reset]") {
+    reset_utils::Partition partition{"realm_id", random_string(20)};
+    Property partition_prop = {partition.property_name, PropertyType::String | PropertyType::Nullable};
+    Schema schema{
+        {"object",
+         {
+             {"_id", PropertyType::ObjectId, Property::IsPrimary{true}},
+             {"value", PropertyType::Int},
+             partition_prop,
+         }},
+    };
+
+    std::string base_url = get_base_url();
+    REQUIRE(!base_url.empty());
+    auto server_app_config = minimal_app_config(base_url, "c_api_client_reset_tests", schema);
+    server_app_config.partition_key = partition_prop;
+    TestAppSession test_app_session(create_app(server_app_config));
+
+    auto app = test_app_session.app();
+    auto get_valid_config = [&]() -> SyncTestFile {
+        create_user_and_log_in(app);
+        return SyncTestFile(app->current_user(), partition.value, schema);
+    };
+    SyncTestFile local_config = get_valid_config();
+    SyncTestFile remote_config = get_valid_config();
+
+    struct ClientSyncObject {
+        static ClientSyncObject& instance()
```

Comment:
What is the purpose of using a static variable here? You don't reset the state between the manual and discardLocal tests and that seems wrong because it will already be stopped by the time it gets to the second test. Why not just use a local std::atomic_bool instead of this struct?

Reply:
It's subtle, lambdas with captures do not translate to C like functions, the only type of lambda that does that is the one without capture list. This is the reason behind these objects that behave like singletons.

Reply:
Interesting, I did not know this!

Reply:
Interesting, I did not know this!